simulariumio/writers/binary_writer.py

```diff
@@ -0,0 +1,441 @@
+                + BINARY_SETTINGS.SPATIAL_BLOCK_HEADER_CONSTANT_N_VALUES
+                + 2 * (file_chunks[current_chunk].n_frames + 1)
+            )
+            spatial_header_n_bytes = (
+                BINARY_SETTINGS.BYTES_PER_VALUE * spatial_header_n_values
+            )
+            frame_n_values = BINARY_SETTINGS.FRAME_HEADER_N_VALUES + buffer_size
+            frame_n_bytes = BINARY_SETTINGS.BYTES_PER_VALUE * frame_n_values
+            if frame_n_bytes > max_spatial_bytes:
+                raise Exception(
+                    f"Frame {frame_index} is too large for a simularium file "
+                    f"({frame_n_bytes} bytes), try filtering out some data."
+                )
+            new_bytes = spatial_header_n_bytes + current_frames_bytes + frame_n_bytes
+            if new_bytes > max_spatial_bytes:
+                file_chunks.append(BinaryChunk(frame_index))
+                current_chunk += 1
+                current_frames_bytes = 0
+            file_chunks[current_chunk].n_frames += 1
+            file_chunks[current_chunk].n_values += frame_n_values
+            file_chunks[current_chunk].frame_n_values.append(frame_n_values)
+            current_frames_bytes += frame_n_bytes
+        for chunk in file_chunks:
+            chunk.n_bytes = BINARY_SETTINGS.BYTES_PER_VALUE * (
+                BINARY_SETTINGS.BLOCK_HEADER_N_VALUES
+                + BINARY_SETTINGS.SPATIAL_BLOCK_HEADER_CONSTANT_N_VALUES
+                + 2 * chunk.n_frames  # frame offsets and lengths
+                + chunk.n_values
+            )
+        return file_chunks, traj_info_n_bytes, plot_data_n_bytes
+
+    @staticmethod
+    def _binary_header(
+        traj_info_n_bytes: int,
+        spatial_data_n_bytes: int,
+        plot_data_n_bytes: int,
+    ) -> BinaryValues:
+        """
+        Return the binary header values and format
+        """
+        header_n_bytes = BinaryWriter._header_n_bytes()
+        header_format = (
+            f"<{len(BINARY_SETTINGS.FILE_IDENTIFIER)}s"
+            f"{BINARY_SETTINGS.HEADER_N_INT_VALUES}I"
+        )
+        block_types = BINARY_SETTINGS.DEFAULT_BLOCK_TYPES
+        block_n_bytes = [traj_info_n_bytes, spatial_data_n_bytes, plot_data_n_bytes]
+        block_offsets = [
+            header_n_bytes,
+            header_n_bytes + block_n_bytes[0],
+            header_n_bytes + block_n_bytes[0] + block_n_bytes[1],
+        ]
+        return BinaryValues(
+            values=(
+                [bytes(BINARY_SETTINGS.FILE_IDENTIFIER, "utf-8")]
+                + [header_n_bytes, BINARY_SETTINGS.VERSION, BINARY_SETTINGS.N_BLOCKS]
+                + [
+                    block_offsets[0],
+                    block_types[0],
+                    block_n_bytes[0],
+                    block_offsets[1],
+                    block_types[1],
+                    block_n_bytes[1],
+                    block_offsets[2],
+                    block_types[2],
+                    block_n_bytes[2],
```

Comment:
Just for fun, I googled "interleave multiple lists in python" and found this:
```
[val for tup in zip(block_offsets, block_types, block_n_bytes) for val in tup]
```
assuming the lists are all the same length... which would be future-proof for arbitrary numbers of blocks.

Reply:
cool!